core/domain/story_services.py

```diff
@@ -649,20 +649,27 @@ def publish_story(story_id, committer_id):
         Exception. The story is already published.
         Exception. The user does not have enough rights to publish the story.
     """
-    def _is_node_valid_for_publishing(node):
-        if not node.exploration_id:
-            raise Exception(
-                'Story node does not contain an exploration id.')
-        if not exp_services.get_exploration_by_id(
-                node.exploration_id, strict=False):
-            raise Exception(
+    def _are_nodes_valid_for_publishing(story_nodes):
+        exploration_id_list = []
+        for node in story_nodes:
+            if not node.exploration_id:
+                raise Exception(
+                    'Story node with id %s does not contain an '
+                    'exploration id.' % node.id)
+            exploration_id_list.append(node.exploration_id)
+        for exploration in  exp_services.get_multiple_explorations_by_id(
+            exploration_id_list):
+            if not exploration:
+                raise Exception(
                 'Exploration id %s doesn\'t exist.' % node.exploration_id)
```

Comment:
Here and below, why are you referring to `node` outside its for loop?

Reply:
Changed